openhgnn/sampler/test_MAGNN_sampler.py

```diff
@@ -0,0 +1,209 @@
+        th.save(model.state_dict(), self.save_path)
+        self.val_loss_min = val_loss
+
+def load_hg(args):
+    hg_dir = 'openhgnn/dataset/'
+    hg,_ = dgl.load_graphs(hg_dir+'{}/graph.bin'.format(args.dataset), [0])
+    hg = hg[0]
+    return hg
+
+def svm_test(X, y, test_sizes=(0.2, 0.4, 0.6, 0.8), repeat=10):
+    # This method is implemented by author
+    random_states = [182318 + i for i in range(repeat)]
+    result_macro_f1_list = []
+    result_micro_f1_list = []
+    for test_size in test_sizes:
+        macro_f1_list = []
+        micro_f1_list = []
+        for i in range(repeat):
+            X_train, X_test, y_train, y_test = train_test_split(
+                X, y, test_size=test_size, shuffle=True, random_state=random_states[i])
+            svm = LinearSVC(dual=False)
+            svm.fit(X_train, y_train)
+            y_pred = svm.predict(X_test)
+            macro_f1 = f1_score(y_test, y_pred, average='macro')
+            micro_f1 = f1_score(y_test, y_pred, average='micro')
+            macro_f1_list.append(macro_f1)
+            micro_f1_list.append(micro_f1)
+        result_macro_f1_list.append([np.mean(macro_f1_list), np.std(macro_f1_list)])
+        result_micro_f1_list.append([np.mean(micro_f1_list), np.std(micro_f1_list)])
+    return result_macro_f1_list, result_micro_f1_list
+
+if __name__ == '__main__':
+
+    warnings.filterwarnings('ignore')
+    args = argparse.Namespace(**CONFIG)
+    hg = load_hg(args)
+    train_mask = hg.nodes[args.category].data['train_mask']
+    val_mask = hg.nodes[args.category].data['val_mask']
+    test_mask = hg.nodes[args.category].data['test_mask']
+    model = MAGNN.build_model_from_args(args, hg)
+    print(args)
+
+    sampler = MAGNN_sampler(g=hg, mask=train_mask.cpu().numpy(), n_layers=args.num_layers, category=args.category,
+                            metapath_list=model.metapath_list, num_samples=args.num_samples, dataset_name=args.dataset)
+
+    dataloader = DataLoader(dataset=sampler, batch_size=args.batch_size, shuffle=True, num_workers=args.num_workers,
+                            collate_fn=collate_fn, drop_last=False)
+
+    optimizer = th.optim.Adam(model.parameters(), lr=args.lr, weight_decay=args.weight_decay)
+    early_stop = EarlyStopping(patience=args.patience, verbose=True,
+                               save_path='/home/ubuntu/openhgnn/openhgnn/sampler/checkpoint/checkpoint_dblp.pt')
+    model = model.to(args.device)
+
+    for epoch in range(args.max_epoch):
+        sampler.mask = train_mask
+        t = time.perf_counter()
+        model.train()
+        print("...Start the mini batch training...")
+        for num_iter, (sub_g, mini_mp_inst, seed_nodes) in enumerate(dataloader):
+            print("Sampling {} seed_nodes with duration(s): {}".format(len(seed_nodes[args.category]), time.perf_counter() - t))
+            model.mini_reset_params(mini_mp_inst)
+            sub_g = sub_g.to(args.device)
+            pred, _ = model(sub_g)
```

Comment:
Why not passing `mini_mp_inst` in the forward function?

Reply:
We don't need to pass `mini_mp_inst` in the forward function. In L125 we call `mini_reset_params()` with parameter `mini_mp_inst` so as to pass this param to the model. Meanwhile, we also reset some other important parameters in calling this method.